internal/gatewayapi/runner/runner.go

```diff
@@ -428,10 +428,11 @@ func (r *Runner) getAllStatuses() *StatusesToDelete {
 }
 
 func (r *Runner) deleteStatusKeys(ds *StatusesToDelete) {
-	for key := range ds.GatewayStatusKeys {
-		r.ProviderResources.GatewayStatuses.Delete(key)
-		delete(ds.GatewayStatusKeys, key)
-	}
+	// Do not delete the status keys for the Gateway because the Gateway status has also been stored into the ProviderResources
```

Comment:
an assumption we are making here is that the watchable will first kick and update status and then a service spec will be updated which will result in updating the address. These steps although are ordered in theory, the ordering cannot be guaranteed in the reconciler, which may cause issues related to stale values being set

Reply:
this logic is going to cause issues

I suggest reverting some of this and instead setting the value only if set i.e. if `Status.Address` is non zero value, then set it, or if `Status.Conditions` is non zero value, then set it. This ensures that the gateway api path doesnt update `Addresses`

Reply:
the gateway api path won't update `Address` or programming status, it's handled in the `updateStatusForGateway` of the status updater. https://github.com/envoyproxy/gateway/blob/dd1c66fb024e1ce76f59fe1480900307e9ae480d/internal/provider/kubernetes/status.go#L524-L548